systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaNodePoolST.java

```diff
@@ -0,0 +1,162 @@
+     * 3. - Produce and consume messages in given Kafka Cluster.
+     * - Clients can produce and consume messages.
+     * 4. - Modify Kafka custom resource annotation strimzi.io/node-pool to disable management by KafkaNodePool.
+     * - new StatefulSet is created, replacing former one, Pods are replaced and specification from KafkaNodePool (i.e., changed replica count) are ignored.
+     * 5. - Produce and consume messages in given Kafka Cluster.
+     * - Clients can produce and consume messages.
+     * 6. - Modify Kafka custom resource annotation strimzi.io/node-pool to enable management by KafkaNodePool.
+     * - new StatefulSet is created, replacing former one, Pods are replaced and specification from KafkaNodePool (i.e., changed replica count) has priority over Kafka specification.
+     * 7. - Produce and consume messages in given Kafka Cluster.
+     * - Clients can produce and consume messages.
+     *
+     * @usecase
+     * - kafka-node-pool
+     */
+    @ParallelNamespaceTest
+    void testKafkaManagementTransferToAndFromKafkaNodePool(ExtensionContext extensionContext) {
+
+        final TestStorage testStorage = new TestStorage(extensionContext);
+
+        // setup clients
+        KafkaClients clients = new KafkaClientsBuilder()
+            .withProducerName(testStorage.getProducerName())
+            .withConsumerName(testStorage.getConsumerName())
+            .withBootstrapAddress(KafkaResources.plainBootstrapAddress(testStorage.getClusterName()))
+            .withTopicName(testStorage.getTopicName())
+            .withMessageCount(testStorage.getMessageCount())
+            .withDelayMs(200)
+            .withNamespaceName(clusterOperator.getDeploymentNamespace())
+            .build();
+
+        LOGGER.info("Deploying Kafka Cluster: {}/{} controlled by KafkaNodePool: {}", testStorage.getNamespaceName(), testStorage.getClusterName(), testStorage.getKafkaNodePoolName());
+        resourceManager.createResource(extensionContext, KafkaTemplates.kafkaPersistent(testStorage.getClusterName(), 3, 1).build());
+        LOGGER.info("Kafka Cluster: {}/{} is ready", testStorage.getNamespaceName(), testStorage.getClusterName());
+
+        // increase number of kafka replicas in KafkaNodePool
+        LOGGER.info("Modifying KafkaNodePool: {}/{} by increasing number of Kafka replicas from '3' to '4'", testStorage.getNamespaceName(), testStorage.getKafkaNodePoolName());
+        KafkaNodePoolResource.replaceKafkaNodePoolResourceInSpecificNamespace(testStorage.getKafkaNodePoolName(), kafkaNodePool -> {
+            kafkaNodePool.getSpec().setReplicas(4);
+        }, testStorage.getNamespaceName());
+
+        waitForPodSetKafkaPodsReady(testStorage.getNamespaceName(), 4, testStorage.getKafkaStatefulSetName());
+
+        LOGGER.info("Producing and Consuming messages with clients: {}, {} in Namespace {}", testStorage.getProducerName(), testStorage.getConsumerName(), testStorage.getNamespaceName());
```

Comment:
Wouldn't be more interesting to produce and consume messages during scaling up the KafkaNodePool replicas and do the check after such an operation is done? :) @im-konge @henryZrncik ?

Reply:
There is (IIRC) a complete wipe of the resource, there is different bootstrapName etc. when you are switching between nodepools enabled and disabled.
So I guess this is not possible. But maybe I'm wrong.